src/arduino/arduinoContentProvider.ts

```diff
@@ -50,7 +47,7 @@ export class ArduinoContentProvider implements vscode.TextDocumentContentProvide
         this.addHandlerWithLogger("load-examples", "/api/examples", async (req, res) => await this.getExamples(req, res));
         this.addHandlerWithLogger("open-example", "/api/openexample", (req, res) => this.openExample(req, res), true);
 
-        this._webserver.start();
+        await this._webserver.start();
```

Comment:
FYI - you can just `return this._webserver.start();` and return the promise (and then you don't have to await here ) and you don't have to mark this method `async`